backend.go

```diff
@@ -87,81 +94,223 @@ func (b *azureSecretBackend) periodicFunc(ctx context.Context, sys *logical.Requ
 		!replicationState.HasState(consts.ReplicationPerformanceStandby) {
 
 		b.Logger().Debug("starting periodic func")
-		if !b.updatePassword {
-			b.Logger().Debug("periodic func", "rotate-root", "no rotate-root update")
-			return nil
-		}
 
-		config, err := b.getConfig(ctx, sys.Storage)
+		err := b.rotateRootCreds(ctx, sys.Storage)
 		if err != nil {
+			b.Logger().Error("rotate root credentials failed", "error", err)
```

Comment:
should failing to rotate root creds stop us from performing SP refresh? Or is it that, if we can't rotate the root creds were are busted enough that it's not worth trying?

Reply:
If we couldn't rotate the root creds I figure things are broken enough that it isn't worth trying to rotate other creds.